admin-ui/src/lib/locale-preference.tsx

```
@@ -0,0 +1,53 @@
+import { createContext, useCallback, useContext, useEffect, useMemo, useState } from "react"
+
+import { i18n, LOCALE_STORAGE_KEY, type SupportedLocale, readLocalePreference } from "@/lib/i18n"
+
+function writeLocalePreference(value: SupportedLocale): void {
+  try {
+    window.localStorage.setItem(LOCALE_STORAGE_KEY, value)
```

ESLint

[error] Prefer `globalThis` over `window`.

(unicorn/prefer-global-this)